nomad/encrypter.go

```diff
@@ -368,6 +368,13 @@ func (e *Encrypter) AddUnwrappedKey(rootKey *structs.UnwrappedRootKey, isUpgrade
 // but it returns an error for ease of testing.
 func (e *Encrypter) AddWrappedKey(ctx context.Context, wrappedKeys *structs.RootKey) error {
 
+	// If the passed root key does not contain any wrapped keys, it has no
+	// decryption tasks to perform and therefore should not supersede or impact
+	// tasks running for the same keyID.
```

Comment:
I might add a note here about the limited circumstances we expect this to be true: when FSM applying / restoring from snapshot the metadata for a legacy key. When we remove backcompat for legacy keys entirely (1.12.0), we can potentially remove or at least review this again.